mmaction/datasets/pipelines/pose_loading.py

```diff
@@ -0,0 +1,666 @@
+
+        if 'frame_inds' not in results:
+            results['frame_inds'] = np.arange(results['total_frames'])
+
+        if results['frame_inds'].ndim != 1:
+            results['frame_inds'] = np.squeeze(results['frame_inds'])
+
+        offset = results.get('offset', 0)
+        frame_inds = results['frame_inds'] + offset
+
+        if 'kpscore' in results:
+            kpscore = results['kpscore']
+            if self.random_drop:
+                self._drop_kpscore(kpscore)
+
+            results['kpscore'] = kpscore[:, frame_inds].astype(np.float32)
+
+        if 'kp' in results:
+            results['kp'] = results['kp'][:, frame_inds].astype(np.float32)
+
+        return results
+
+    def __repr__(self):
+        repr_str = (f'{self.__class__.__name__}('
+                    f'random_drop={self.random_drop}, '
+                    f'random_seed={self.random_seed}, '
+                    f'drop_prob={self.drop_prob}, '
+                    f'manipulate_joints={self.manipulate_joints})')
+        return repr_str
+
+
+@PIPELINES.register_module()
+class LoadKineticsPose:
+    """Load Kinetics Pose given filename (The format should be pickle)
+
+    Required keys are "filename", "total_frames", "img_shape", "frame_inds",
+    "anno_inds" (for mmpose source, optional), added or modified keys are "kp",
+    "kpscore".
+
+    Args:
+        io_backend (str): IO backend where frames are stored. Default: 'disk'.
+        squeeze (bool): Whether to remove frames with no human pose.
+            Default: True.
+        max_person (int): The max number of persons in a frame. Default: 10.
+        keypoint_weight (dict): The weight of keypoints. We set the confidence
+            score of a person as the weighted sum of confidence scores of each
+            joint. Persons with low confidence scores are dropped (if exceed
+            max_person). Default: dict(face=1, torso=2, limb=3).
+        source (str): The sources of the keypoints used. Choices are 'mmpose'
+            and 'openpose'. Default: 'mmpose'.
+        kwargs (dict, optional): Arguments for FileClient.
+    """
+
+    # squeeze (Remove those frames that w/o. keypoints)
+    # kp2keep (The list of keypoint ids to keep)
+    def __init__(self,
+                 io_backend='disk',
+                 squeeze=True,
+                 max_person=100,
+                 keypoint_weight=dict(face=1, torso=2, limb=3),
+                 source='mmpose',
+                 **kwargs):
+
+        self.io_backend = io_backend
+        self.squeeze = squeeze
+        self.max_person = max_person
+        self.keypoint_weight = keypoint_weight
+        self.source = source
+
+        if source == 'openpose':
+            self.kpsubset = dict(
+                face=[0, 14, 15, 16, 17],
+                torso=[1, 2, 8, 5, 11],
+                limb=[3, 4, 6, 7, 9, 10, 12, 13])
+        elif source == 'mmpose':
+            self.kpsubset = dict(
```

Comment:
Which file in mmpose defines this

Reply:
No, it just indicates that the pose annotations are extracted by mmpose.